pipelinerl/async_vlm.py

```diff
@@ -0,0 +1,218 @@
+import logging
+import aiohttp
+from typing import Optional
+import torch
+import base64
+import io
+
+from pipelinerl.finetune.data import MASKED_TOKEN_ID
+
+from tapeagents.core import LLMCall, LLMOutput, Prompt, TokenLogprob, TrainingText
+from tapeagents.llms.trainable import TrainableLLM
+from transformers import AutoProcessor
+from PIL import Image
+
+logger = logging.getLogger(__name__)
+
+
+class VLMCall(LLMCall):
+    """Extended LLMCall with vision-language model metrics."""
+    
+    num_images: int = 0
```

Comment:
I don't think we need this extra structure. The extraction of images from `messages` can occur downstream at the time when we make `TrainingText`.